api/transactions.js

```diff
@@ -239,37 +239,20 @@ function getTransaction(api, account, identifier, requestOptions, callback) {
   assert.strictEqual(typeof requestOptions, 'object');
 
   var options = { };
+  validate.address(account, true);
+  validate.paymentIdentifier(identifier);
+  validate.ledger(requestOptions.ledger, true);
 
-  function validateOptions(async_callback) {
-    if (account && !ripple.UInt160.is_valid(account)) {
-      return callback(new errors.InvalidRequestError(
-        'Invalid parameter: account. Must be a valid Ripple Address'));
-    }
-
-    if (!_.isString(identifier)) {
-      return callback(new errors.InvalidRequestError(
-        'Missing parameter: identifier'));
-    }
-
-    if (!_.isUndefined(requestOptions.ledger)) {
-      if (isNaN(requestOptions.ledger)) {
-        return callback(new errors.InvalidRequestError(
-          'Invalid parameter: ledger. Must be a number'));
-      } else if (api.remote.getServer()
-                 && !api.remote.getServer().hasLedger(requestOptions.ledger)) {
-        return callback(new errors.NotFoundError('Ledger not found'));
-      }
-    }
+  if (validator.isValid(identifier, 'Hash256')) {
+    options.hash = identifier;
+  } else {
+    options.client_resource_id = identifier;
+  }
 
-    if (validator.isValid(identifier, 'Hash256')) {
-      options.hash = identifier;
-      async_callback();
-    } else if (validator.isValid(identifier, 'ResourceId')) {
-      options.client_resource_id = identifier;
-      async_callback();
-    } else {
-      return callback(new errors.InvalidRequestError('Parameter not a valid '
-        + 'transaction hash or client_resource_id: identifier'));
+  if (!_.isUndefined(requestOptions.ledger)) {
```

Comment:
This refactoring is mostly trying to preserve existing behavior. Later the goal is to use JSON schema, so all these individual checks will be gone.